packages/utils/blueprint-cli/src/publish/codecatalyst-authentication.ts

```diff
@@ -25,55 +26,70 @@ export function generateHeaders(authentication: CodeCatalystAuthentication, iden
   }
 }
 
-export const codecatalystAuthentication = async (log: pino.BaseLogger, options: {
-  cookie?: string;
-  endpoint: string;
-  region?: string;
-}): Promise<CodeCatalystAuthentication | undefined> => {
-
+export const codecatalystAuthentication = async (
+  log: pino.BaseLogger,
+  options: {
+    cookie?: string;
+    endpoint: string;
+    region?: string;
+  },
+): Promise<CodeCatalystAuthentication | undefined> => {
   if (options.cookie) {
     return {
       type: 'cookie',
       token: options.cookie,
     };
-  }
-  log.warn('Attempting to fetch credentials from AWS CLI profile.');
-  try {
-    const codecatalyst = new CodeCatalystClient({
-      region: options.region,
-      endpoint: path.join('https://', options.endpoint),
-    });
+  } else if (process.env.AWS_CONTAINER_TOKEN_ENDPOINT) {
+    log.warn('Attempting to fetch credentials from the CodeCatalyst workflow action environment.');
+    try {
+      const response = await axios.default.get(process.env.AWS_CONTAINER_TOKEN_ENDPOINT);
+      const credentials = response.data as { AccessKeyId: string; SecretAccessKey: string };
+      return {
+        type: 'workflowtoken',
+        token: `${credentials.AccessKeyId}:${credentials.SecretAccessKey}`,
+      };
+    } catch (e: any) {
+      log.error(e);
+      log.error('Unable to fetch credentials from the CodeCatalyst workflow action environment.');
+    }
+  } else {
+    log.warn('Attempting to fetch credentials from AWS CLI profile.');
```

Comment:
Probably unlikely, but could there be a situation where `AWS_CONTAINER_TOKEN_ENDPOINT` is defined but returns some invalid value? I think in that case we would want to fall back to the authing from the CLI profile.

Reply:
I considered it, but thought it would be strange given that our current authorization mechanisms don't fall through.  I.e. if we went forward with that approach, we'd probably want cookie auth to fall through to token, then token to fall through to SSO?

Reply:
That's true, I think a fall through could be somewhat useful here since we don't exactly give the user much of a choice in what authentication mechanism they want to use. But the chances that someone has an `AWS_CONTAINER_TOKEN_ENDPOINT` defined and wants to use SSO auth is probably going to be pretty slim. I meant to mark my comment as non-blocking, I think this is good how it is.

Reply:
We could perhaps tie it to something more specific to CodeCatalyst Workflows, perhaps a different environment variable?

Reply:
Actually it looks like `AWS_CONTAINER_TOKEN_ENDPOINT` *is* something specific to workflows, despite the name. I think change is safe as it gets.